cmd/thanos/store.go

```diff
@@ -363,14 +363,21 @@ func runStore(
 
 			level.Info(logger).Log("msg", "initializing bucket store")
 			begin := time.Now()
-			if err := bs.InitialSync(ctx); err != nil {
+
+			err := runutil.RetryWithLog(logger, 2*time.Second, ctx.Done(), func() error {
+				syncErr := bs.InitialSync(ctx)
+				return errors.Wrap(syncErr, "bucket store initial sync")
```

Comment:
We don't need to wrap the error here. Otherwise in line 374 we will wrap the error twice.